sw/device/silicon_creator/lib/boot_svc/boot_svc_empty.c

```diff
@@ -0,0 +1,22 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/silicon_creator/lib/boot_svc/boot_svc_empty.h"
+
+#include "sw/device/silicon_creator/lib/drivers/rnd.h"
+
+void boot_svc_empty_init(boot_svc_empty_t *msg) {
+  // We use `uint32_t` instead of `size_t` so that end-of-loop check passes both
+  // on- and off-target tests.
+  uint32_t i = 0, j = kBootSvcEmptyPayloadWordCount - 1;
+  for (; launder32(i) < kBootSvcEmptyPayloadWordCount &&
+         launder32(j) < kBootSvcEmptyPayloadWordCount;
+       ++i, --j) {
+    msg->rand_data[i] = rnd_uint32();
```

Comment:
Just curious: is there a (security) reason init with random data instead of just all 0s or 1s?

Reply:
The retention RAM scrambling key persists across resets so that we can access its contents. Also, retention RAM is initialized using the LFSR only after a PoR. Clearing with random data is more of a defense in depth for reducing leakage of potentially confidential information. This also limits the reproducibility and scale of any possible attack.

@moidx PLMK if you have any comments.